projects/distributed-tracing/src/shared/dashboard/widgets/waterfall/waterfall/waterfall-chart.ts

```diff
@@ -23,6 +23,7 @@ export interface WaterfallData {
   responseBody?: string;
   tags: Dictionary<unknown>;
   errorCount: number;
+  logEvents?: LogEvent[];
```

Comment:
Sort of the same question. Should this be optional? I guess in this case maybe we could expect `logEvents` to be undefined if none are associated with the span?

Reply:
Yeah, this can be the case.

Reply:
logEvents will never return null, always an empty array - unless we're conditionally requesting it (but I'd argue in this context, we should ensure it's there for simplicity downstream).

Reply:
Added extra condition to handle that.
So it is not optional anymore.